staging/src/k8s.io/apimachinery/pkg/runtime/serializer/json/json.go

```diff
@@ -213,6 +241,20 @@ func (s *Serializer) Decode(originalData []byte, gvk *schema.GroupVersionKind, i
 		return nil, actual, err
 	}
 
+	// Handle strict mode.
+	if s.strict {
+		// Delayed failure for the strict YAML to JSON conversation,
+		// so that the error output also includes GVK.
+		if errorYAMLToJSON != nil {
+			return nil, actual, fmt.Errorf("%#v: %v", actual, errorYAMLToJSON)
+		}
+		// Return an error for unknown JSON fields in strict mode.
```

Comment:
nit: "Return an error if there are fields in the provided JSON data that do not exist in the object, or if there are duplicated fields in the JSON file"